pkg/secret/secret.go

```diff
@@ -0,0 +1,165 @@
+	secrets, err := ss.client.List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+	return secrets.Items, nil
+}
+
+// GetUnused returns unused resources
+func (ss SecretsService) GetUnused(namespace string, resources []v1.Secret) (unusedResources []v1.Secret, funcErr error) {
+	var usedSecrets []v1.Secret
+	funk.ForEach(openshift.PredefinedResources, func(predefinedResource schema.GroupVersionResource) {
+		funk.ForEach(resources, func(secret v1.Secret) {
+
+			secretName := secret.GetName()
+
+			if funk.Contains(usedSecrets, secret) {
+				// already marked as existing, skip this
+				return
+			}
+			contains, err := ss.helper.ResourceContains(namespace, secretName, predefinedResource)
+			if err != nil {
+				funcErr = err
+				return
+			}
+
+			if contains {
+				usedSecrets = append(usedSecrets, secret)
+			}
+		})
+	})
+
+	for _, resource := range resources {
+		if !funk.Contains(usedSecrets, resource) {
+			unusedResources = append(unusedResources, resource)
+		}
+	}
+
+	return unusedResources, funcErr
+}
+
+// Delete removes secrets
+func (ss SecretsService) Delete(secrets []v1.Secret) {
+	for _, resource := range secrets {
+		namespace := resource.GetNamespace()
+		kind := "Secret"
+		name := resource.GetName()
+
+		log.Infof("Deleting %s %s/%s", kind, namespace, name)
+
+		err := ss.client.Delete(name, &metav1.DeleteOptions{})
+
+		if err != nil {
+			log.WithError(err).Errorf("Failed to delete %s %s/%s", kind, namespace, name)
+		}
+	}
+}
+
+//FilterByTime returns secrets which are older than specified date
+func (ss SecretsService) FilterByTime(secrets []v1.Secret, olderThan time.Time) (filteredResources []v1.Secret) {
+	log.WithFields(log.Fields{
+		"olderThan": olderThan,
+		"resources": secrets,
+	}).Debug("Filtering resources older than the specified time")
+
+	for _, resource := range secrets {
+		lastUpdatedDate := resource.GetCreationTimestamp().Time
+		if lastUpdatedDate.Before(olderThan) {
+			filteredResources = append(filteredResources, resource)
+		} else {
+			name := resource.GetName()
+			log.WithField("name", name).Debugf("Filtered resource %s", name)
+		}
+	}
+
+	return filteredResources
+}
+
+// FilterByMaxCount keep at most n newest resources. The list of secrets is sorted in descending ordered in
+func (ss SecretsService) FilterByMaxCount(secrets []v1.Secret, keep int) (filteredResources []v1.Secret) {
+
+	log.WithFields(log.Fields{
+		"keep":    keep,
+		"secrets": secrets,
+	}).Debug("Filtering ordered by time Resources from the n'th number specified")
+
+	sort.SliceStable(secrets, func(i, j int) bool {
+		return secrets[j].GetCreationTimestamp().Time.Before(secrets[i].GetCreationTimestamp().Time)
+	})
+
+	if len(secrets) <= keep {
+		return []v1.Secret{}
+	}
+
+	return secrets[keep:]
+}
+
+// Print prints the given resource line by line. In batch mode, only the resource is printed, otherwise default
+// log with info level
+func (ss SecretsService) Print(resources []v1.Secret, batch bool) {
```

Comment:
It may make more sense to pass the batch flag in other means, e.g. pass a configuration struct in the constructor? The Delete() method may also need the batch flag, or to configure the DeleteOptions{}

Reply:
I could wrap it in an Option structure but I wouldn't include it in the Service.

Reply:
Actually a config structure wouldn't be bad